pkg/operator/ceph/csi/csi.go

```diff
@@ -40,10 +41,22 @@ func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage
 	}
 
 	if CSIEnabled() {
-		if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil {
-			logger.Errorf("failed to start Ceph csi drivers. %v", err)
-			return
+		csiLock.Lock()
```

Comment:
To guarantee it is freed when the method returns...
```suggestion
		csiLock.Lock()
		defer csiLock.Unlock()
```